dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java

```diff
@@ -100,7 +80,7 @@ private long getWorkflowId(long workflowVersionId) {
         }
 
         long workflowVersionId = workflowVersion.getId();
-        long workflowId = getWorkflowId(workflowVersionId);
+        long workflowId = AliasHelper.getWorkflowId(workflowDAO, workflowVersionId);
         Workflow workflow = workflowDAO.findById(workflowId);
```

Comment:
AliasHelper.getWorkflowId already does a database lookup, and then you take the id from there and do the same database lookup again. Database lookups are relatively expensive and should try to be avoided if possible. Without fully going over all usages, maybe you should just return a Workflow instead of an id from AliasHelper.getWorkflowId.

Reply:
Now return a Workflow instead of an id from AliasHelper.getWorkflow